core/connector/src/main/java/org/exoplatform/wcm/connector/collaboration/OpenInOfficeConnector.java

```diff
@@ -216,26 +234,38 @@ public Response checkout(@Context Request request,
    */
   @GET
   @Path("/{linkFilePath}/")
+  @RolesAllowed("users")
   @Produces("application/internet-shortcut")
   public Response createShortcut(@Context HttpServletRequest httpServletRequest,
                            @PathParam("linkFilePath") String linkFilePath,
                            @QueryParam("filePath") String filePath,
                            @QueryParam("workspace") String workspace
   ) throws Exception {
-    Session session = WCMCoreUtils.getSystemSessionProvider().getSession(workspace, WCMCoreUtils.getRepository());
-    Node node = (Node)session.getItem(filePath);
-    String repo = WCMCoreUtils.getRepository().getConfiguration().getName();
-
-    String obsPath = httpServletRequest.getScheme()+ "://" + httpServletRequest.getServerName() + ":"
-            +httpServletRequest.getServerPort() + "/"
-            + WCMCoreUtils.getRestContextName()+ "/private/jcr/" + repo + "/" + workspace + node.getPath();
-
-    String shortCutContent = "[InternetShortcut]\n";
-    shortCutContent+="URL="+obsPath+"\n";
-    return Response.ok(IOUtils.toInputStream(shortCutContent), MediaType.APPLICATION_OCTET_STREAM)
-            .header("Content-Disposition","attachment; filename="+node.getName()+".url")
-            .header("Content-type", "application/internet-shortcut")
-            .build();
+    try {
+      Node node = null;
+      try {
+        node = (Node) nodeFinder.getItem(workspace, filePath);
+      } catch (PathNotFoundException e) {
+        node = (Node) nodeFinder.getItem(workspace, Text.unescapeIllegalJcrChars(filePath));
+      }
+      String repo = WCMCoreUtils.getRepository().getConfiguration().getName();
+
+      String obsPath = httpServletRequest.getScheme() + "://" + httpServletRequest.getServerName() + ":"
```

Comment:
Shouldn't this be /portal/rest instead of /rest ?

Reply:
Nope, this is needed with Basic authentication